staging/src/k8s.io/apiserver/pkg/storage/interfaces.go

```diff
@@ -160,7 +161,10 @@ type Interface interface {
 	// (e.g. reconnecting without missing any updates).
 	// If resource version is "0", this interface will get current object at given key
 	// and send it in an "ADDED" event, before watch starts.
-	Watch(ctx context.Context, key string, resourceVersion string, p SelectionPredicate) (watch.Interface, error)
+	// The 'timeout' indicates the duration in which the client keeps watch. The server
+	// will send bookmark events to the client within the duration, and close the watch
```

Comment:
The server isn't obliged to send bookmark events - I would remove this sentence (or reformulate it so that it's clear that server may but doesn't have to send it) - something similar to: https://github.com/kubernetes/kubernetes/pull/74074/files#diff-eca3b8d856fa2e661f6da91b61de5e76R353